File: app/src/main/java/com/seafile/seadroid2/framework/model/sdoc/FileTagWrapperModel.java
package com.seafile.seadroid2.framework.model.sdoc;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.List;

public class FileTagWrapperModel implements Parcelable {
    public List<MetadataModel> metadata;
    public List<RecordResultModel> results;

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeTypedList(this.metadata);
        dest.writeTypedList(this.results);
    }

    public FileTagWrapperModel() {
    }

    protected FileTagWrapperModel(Parcel in) {
        this.metadata = in.createTypedArrayList(MetadataModel.CREATOR);
        this.results = in.createTypedArrayList(RecordResultModel.CREATOR);
    }

    public static final Creator<FileTagWrapperModel> CREATOR = new Creator<FileTagWrapperModel>() {
        @Override
        public FileTagWrapperModel createFromParcel(Parcel source) {
            return new FileTagWrapperModel(source);
        }

        @Override
        public FileTagWrapperModel[] newArray(int size) {
            return new FileTagWrapperModel[size];
        }
    };
}

File: app/src/main/java/com/seafile/seadroid2/framework/model/sdoc/FileTagWrapperModel.kt
package com.seafile.seadroid2.framework.model.sdoc

import android.os.Parcel
import android.os.Parcelable

class FileTagWrapperModel() : Parcelable {
    @JvmField
    var metadata: MutableList<MetadataModel>? = null

    @JvmField
    var results: MutableList<FileTagResultModel>? = null

    override fun describeContents(): Int = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeTypedList(metadata)
        dest.writeTypedList(results)
    }

    private constructor(parcel: Parcel) : this() {
        metadata = parcel.createTypedArrayList(MetadataModel.CREATOR)
        results = parcel.createTypedArrayList(FileTagResultModel.CREATOR)
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<FileTagWrapperModel> =
            object : Parcelable.Creator<FileTagWrapperModel> {
                override fun createFromParcel(source: Parcel): FileTagWrapperModel = FileTagWrapperModel(source)
                override fun newArray(size: Int): Array<FileTagWrapperModel?> = arrayOfNulls(size)
            }
    }
}
